src/main/java/com/minecolonies/coremod/util/InventoryFunctions.java

```diff
@@ -78,51 +78,56 @@ public static boolean matchFirstInInventoryWithInventory(
     /**
      * Topmost matchFirst function, will stop after it finds the first itemstack.
      *
-     * @param inventory the inventory to search in
+     * @param provider the provider to search in
      * @param tester    the function to use for testing slots
      * @return true if it found a stack
      */
-    private static boolean matchFirstInInventory(
-                                                  final IInventory inventory, @NotNull final Function<IInventory, Function<Integer,
+    private static boolean matchFirstInProvider(
+                                                  final ICapabilityProvider provider, @NotNull final Function<ICapabilityProvider, Function<Integer,
                                                                                                                             Predicate<ItemStack>>> tester)
     {
-        return matchInInventory(inventory, tester, true);
+        return matchInProvider(provider, tester, true);
     }
 
     /**
-     * Topmost function to actually loop over the inventory.
+     * Topmost function to actually loop over the provider.
      * Will return if it found something.
      *
-     * @param inventory      the inventory to loop over
+     * @param provider      the provider to loop over
      * @param tester         the function to use for testing slots
      * @param stopAfterFirst if it should stop executing after finding one stack that applies
      * @return true if it found a stack
      */
-    private static boolean matchInInventory(
-                                             @Nullable final IInventory inventory,
-                                             @NotNull final Function<IInventory, Function<Integer,
+    private static boolean matchInProvider(
+                                             @Nullable final ICapabilityProvider provider,
+                                             @NotNull final Function<ICapabilityProvider, Function<Integer,
                                                                                            Predicate<ItemStack>>> tester,
                                              final boolean stopAfterFirst)
     {
-        if (inventory == null)
+        if (provider == null)
         {
             return false;
         }
-        final int size = inventory.getSizeInventory();
+
         boolean foundOne = false;
-        for (int slot = 0; slot < size; slot++)
-        {
-            final ItemStack stack = inventory.getStackInSlot(slot);
-            //Unchain the function and apply it
-            if (tester.apply(inventory).apply(slot).test(stack))
+        for(IItemHandler handler : InventoryUtils.getItemHandlersFromProvider(provider)) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'handler' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)